lib/validations.js

```diff
@@ -43,13 +43,21 @@ function isExpectedValidationError(errorResult) {
 
 function isExpectedCase(errorResult, w3cManifestInfo) {
   const errorParams = errorResult.member.split('/').slice(1) //example: /icons/0/purpose -> ['icons', '0', 'purpose']
-  const isIconsPurpose = errorResult.member.includes("icons") && errorResult.member.includes("purpose")
+  const isIconsPurpose = errorResult.member.includes("icons") && errorResult.member.includes("purpose");
+  const isShortcuts = errorResult.member.includes("shortcuts");
+  const isScreenshots = errorResult.member.includes("screenshots"); // in the case we need to extend it to screenshots as well...
 
-  if (isIconsPurpose) {
+  if ((isScreenshots || isShortcuts) && isIconsPurpose) {
+    // don't handle, just let it go through
+    return true;
+
+  } else  if (isIconsPurpose) {
     const [icons, index, purpose] = errorParams
 
-    return w3cManifestInfo.content[icons][index][purpose].split(" ").filter(entry => {
-      return (entry === "any" || entry === "maskable" || entry === "monochrome");
+    return w3cManifestInfo.content.icons
+      && w3cManifestInfo.content.icons[index]
+      && w3cManifestInfo.content.icons[index].purpose.split(" ").filter(entry => {
+        return (entry === "any" || entry === "maskable" || entry === "monochrome");
```

Comment:
I see that this was changed because https://helpzone.me/manifest.json caused this code to throw an error because something was null. What was null, exactly?

Reply:
Oh the check is a guard against future exceptions that kill our service.
The issue was that images within the shortcut were throwing an error when they were space separated lists (e.g. "any maskable").
We were only looking at icons, and since the only use case I see is for icons, hence the skipping of the other two.